lib/redcord/server_scripts/shared/query_helper_methods.erb.lua

```diff
@@ -106,3 +125,58 @@ local function validate_and_parse_query_conditions(hash_tag, model, index_attrs,
   end
   return {index_sets, range_index_sets}
 end
+
+-- Validates that attributes in query are in correct order and range condition is applied only on the last attribute.
+-- Returns a table {index_key, min_string, max_string} to be used for index query.
+local function validate_and_parse_query_conditions_custom(hash_tag, model, index_name, custom_index_attrs, args)
+  if #custom_index_attrs == 0 then
+    error('Index ' .. index_name .. ' does not exist')
+  end
+  local sep = ':'
+  local i = 1
+  local j = 1
+  local min, value_string_min, query_string_min = '', '', ''
+  local max, value_string_max, query_string_max = '', '', ''
+  local is_prev_attr_query_range = false
+  while i <= #args do
+    if is_prev_attr_query_range then
+      error('Range can be applied to the last attribute of query only')
+    end  
+    local attr_key = args[i]
+    if custom_index_attrs[j] == attr_key then
+      min, max = args[i+1], args[i+2]
+      if min then
+        string.gsub (min, '(', '')
+      end
+      if max then
+        string.gsub (max, '(', '')
+      end
+      if j > 1 then
+        query_string_min = query_string_min .. sep
+        query_string_max = query_string_max .. sep
+      else
+        query_string_min = query_string_min .. '['
+        query_string_max = query_string_max .. '['
+      end
+      if min ~= '-inf' then
```

Comment:
do we validate exclusiveness somewhere?

Reply:
I'd expect the following to raise an error:
```
A.where(a: 1, b: Redcord::RangeInterval.new(min: 1, max: 1, max_exclusive: true), index: :first)
```

btw, can we make sure to cover the raise code branches in Lua using test cases?

Reply:
I added validation for exclusiveness in serializer. The example you gave would generate error.